Updating from the main branch

ValterAro opened the pull request.

decodingahmed commented on docs/technical/implementation.md:
```diff
@@ -0,0 +1,91 @@
+# CentOps Implementation
+
+CentOps implemented in .NET 6.0 with C#
```
```suggestion
CentOps is implemented in .NET 6.0 with C#
```

decodingahmed commented on docs/technical/implementation.md:
```diff
@@ -0,0 +1,91 @@
+
+[Automapper](https://automapper.org/) is used to easily map between external types and internal DTOs.  e.g. [CreateUpdateParticipantModel.cs](../../src/CentOps.Api/Models/CreateUpdateParticipantModel.cs) and the internal [ParticipantDto.cs](../../src/CentOps.Api/Services/ModelStore/Models/ParticipantDto.cs).  Very simple mapping between these types is specified in [AutoMapperProfile.cs](../../src ...
+
+[CosmosDB](https://docs.microsoft.com/en-us/azure/cosmos-db/) is used as a data store via implementations of `IParticipantStore` and `IInstitutionStore`.
```
Might be worth mentioning the InMemoryStore implementation also as it is used for tests, I believe?

guypritchard replied:
Yep - I had a few words about this in the Notes section.

decodingahmed commented on docs/technical/implementation.md:
```diff
@@ -0,0 +1,91 @@
+
+[Stryker](https://stryker-mutator.io/) is a mutation testing framework used by this project to ensure tests written to meet coverage requirements are covering crucial areas.
+
+## Implementation And Flow
```
supersupersupernit: for consistency with other headings:
```suggestion
## Implementation and Flow
```

decodingahmed commented on docs/technical/implementation.md:
```diff
@@ -0,0 +1,91 @@
+1. GET request is issued to the admin version of the participants API.
+2. The API Key is validated and calls without the correct API receive 401 (Unauthorized) as a response.
+3. The Admin Participant 'Controller' handles the GET request calling the `Get()` function (also decorated with the `[HttpGet]` attribute to indicate it will handle requests with the HTTP GET Verb)
+4. Dependency Injection will give the Admin Participant Controller an implementation of the `IParticpantStore` interface.  At time of writing this will be a CosmosDb backed implementation of this interface.
```
```suggestion
4. Dependency Injection will give the Admin Participant Controller an implementation of the `IParticipantStore` interface.  At time of writing this will be a CosmosDb backed implementation of this interface.
```

decodingahmed commented on docs/technical/implementation.md:
```diff
@@ -0,0 +1,91 @@
+2. The API Key is validated and calls without the correct API receive 401 (Unauthorized) as a response.
+3. The Admin Participant 'Controller' handles the GET request calling the `Get()` function (also decorated with the `[HttpGet]` attribute to indicate it will handle requests with the HTTP GET Verb)
+4. Dependency Injection will give the Admin Participant Controller an implementation of the `IParticpantStore` interface.  At time of writing this will be a CosmosDb backed implementation of this interface.
+5. The CosmosDB implementation will query the configured data store and returned the participants found within.
```
```suggestion
5. The CosmosDB implementation will query the configured data store and return the participants found within.
```

decodingahmed commented on docs/technical/implementation.md:
```diff
@@ -0,0 +1,91 @@
+
+### ConnectionStrings
+
+Contains the setting for the Admin Api Key.  Current deployment generate a random complex key which is held in Azure Key Vault and piped into CentOps via configuration.
```
```suggestion
Contains the setting for the Admin Api Key.  Current deployments generate a random complex key which is held in Azure Key Vault and piped into CentOps via configuration.
```

decodingahmed commented on docs/technical/implementation.md:
```diff
@@ -0,0 +1,91 @@
+
+Optional functionality is controlled by Feature Toggles.
+
+| Feature Toggle    | Purpose                       |  appSettings.json | yaml environment |
```
```suggestion
| Feature Toggle    | Purpose                       |  appsettings.json | yaml environment |
```

decodingahmed commented on docs/technical/implementation.md:
```diff
@@ -0,0 +1,91 @@
+
+Cross Origin Resource Sharing Settings specified here.
+
+| Setting    | Purpose                       |  appSettings.json | yaml environment |
```
```suggestion
| Setting    | Purpose                       |  appsettings.json | yaml environment |
```

decodingahmed commented on docs/technical/implementation.md:
```diff
@@ -0,0 +1,91 @@
+
+Contains the setting for the Admin Api Key.  Current deployment generate a random complex key which is held in Azure Key Vault and piped into CentOps via configuration.
+
+| Setting    | Purpose                       |  appSettings.json | yaml environment |
```
```suggestion
| Setting    | Purpose                       |  appsettings.json | yaml environment |
```

decodingahmed commented on docs/technical/implementation.md:
```diff
@@ -0,0 +1,91 @@
+
+These settings correspond to the CosmosDb connection used by CentOps.
+
+| Setting    | Purpose                       |  appSettings.json | yaml environment |
```
```suggestion
| Setting    | Purpose                       |  appsettings.json | yaml environment |
```

decodingahmed commented on docs/technical/implementation.md:
```diff
@@ -0,0 +1,91 @@
+
+## Configuration
+
+Configuration of this service can be done in many ways.  [appsettings.json](../../src/CentOps.Api/appsettings.json) is one approach which will be used in local development.  Another is using specifically named environment variables.  Both approaches are documented below.
```
It might be worth mentioning that we're overriding app settings using this technique: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-6.0#naming-of-environment-variables

I don't think it will be immediately obvious what the relationship between appsettings objects and environment variables is to any unfamiliar with it.   What do you think?

guypritchard replied:
Right - this is a good shout.  It's not completely obvious.  I'll add some words.

decodingahmed reviewed (approved):
Some shameless super nits (sorry)

turnerrainer committed:
Update administrative.yml with the get operation for the /admin/participants path (#76) (#77)

Issue #40

Co-authored-by: ValterAro <122787093+ValterAro@users.noreply.github.com>

turnerrainer committed:
#34 update administrative.yml (#80) (#81)

Added GET operation to the /admin/institutions path in the OpenAPI 3.0.3 specification in the administrative.yml file

Co-authored-by: ValterAro <122787093+ValterAro@users.noreply.github.com>

karljogi committed:
18 Require unique id for application request (#31)

* Add docker compose with Ruuter, Resql, Liquibase and DataMapper

* Add GUI dummy code

* Dummy end-to-end example

* Add TIM service

* Ignore tim-db folder

* Implement invitation creation

* Implement unique invitation id saving

* Implement GUI

* Fix create-table-invitation migration

* Create translations, improve error handling

* Clear error once new request is sent

* Add auth mock to invitation creation

* Rename DB table

* Add missing translations

* Fix sql query

* Code styling fixes

* Update invitations DB table with timestamp

* Code styling fixes from PR review

* Create template to validate administrator role

* Move template to more suitable directory

* Create DB table migrations

* Update readme

* Remove implementations from other PR-s

* Update authentication script to match new logic

* Add api endpoint documentation

* Separate public and private ruuter

* Add missing newline

* Initial implementation of checking for unique ID

* Separate public and private GUI

* 18 Require unique ID for application request

* Fix admin template wrong router host

* Implement mock e-mail service

* Add sending mock email to create-invitation DSL

* Remove redundant string cast

* Update API documentation

* Simplify endpoint names

* Update mock service with status code in return

* Split API docs to 3 files

---------

Co-authored-by: baha-a <bhlshrf@gmail.com>

ValterAro committed:
34 get admininstitutions (#83)

* #34 update administrative.yml (#80)

Added GET operation to the /admin/institutions path in the OpenAPI 3.0.3 specification in the administrative.yml file

* Fix administrative.yml

Fixed the issue in administrative.yml OpenAPI spec, where the GET operation for /admin/participants was accidentally deleted

* Update administrative.yml

Fixed the OpenAPI spec to reflect the one in the main branch.

ValterAro committed:
Merge pull request #85 from buerokratt/main

Merging the updates from the main branch

ValterAro committed:
35 updated OpenAPI spec

Added POST operation to the /admin/institutions path and fixed minor errors in operationId and tag
Issue #35

ValterAro committed:
Update administrative.yml

Fixed $ref and description for participants

ValterAro committed:
Merge pull request #86 from buerokratt/35-post-admininstitutions

35 post admin-institutions spec added

ValterAro committed:
Merge pull request #87 from buerokratt/main

Update from the main branch

ValterAro committed:
Update administrative.yml spec

Added specification for GET operation on the admin/institutions/{id} path. Also added admin/institutions tag.

ValterAro committed:
Merge pull request #88 from buerokratt/36-get-admininstitutionsid

36 get admininstitutionsid

ValterAro committed:
Merge pull request #90 from buerokratt/main

Updating from the main branch

ValterAro committed:
Update administrative.yml spec

Added PUT operation for the admin/institutions/{id} path in the administrative.yml OpenAPI 3.0.3 specification.

ValterAro committed:
Merge pull request #91 from buerokratt/37-put-admininstitutionsid

37 put admininstitutionsid spec

ValterAro committed:
Merge pull request #92 from buerokratt/main

Update from the main branch

ValterAro committed:
Update administrative.yml

Added Delete operation for the admin/institutions/{id} path in the administrative.yml OpenAPI 3.0.3 specification.
Issue #38

ValterAro committed:
Merge pull request #93 from buerokratt/38-delete-admininstitutionsid

38 delete admininstitutionsid spec added

ValterAro committed:
Merge pull request #94 from buerokratt/main

Updating from the main branch

ValterAro committed:
Added to the administrative.yml spec

Added **Delete** operation for the **admin/participants/{id}** path in the **administrative.yml** OpenAPI 3.0.3 specification.
#41 issue

ValterAro committed:
Merge pull request #95 from buerokratt/41-delete-adminparticipantsid

41 delete adminparticipantsid spec added
